### ISSUE TITLE
Improve explanation for SC2027

### ISSUE THREAD

**Issue by orbea** [koalaman/shellcheck#1035]
#### For bugs
- Rule Id: `SC2027`
- My shellcheck version: `online`
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

foo=2

echo "1 "$foo"bar"

```

#### Here's what shellcheck currently says:

```
Line 5:
echo "1 "$foo"bar"
         ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
```

#### Here's what I wanted or expected to see:

This warning is correct, however the explanation at https://github.com/koalaman/shellcheck/wiki/SC2027 does not explain why exactly. The wiki could be updated to reflect that the double quotes are being used to make the variable `$foo` rather than `$foobar` and that using braces would be better.

A correct example would be.

```sh

#!/bin/sh

foo=2

echo "1 ${foo}bar"

```

**Comment by Nightfirecat** [comment]
You can feel free to edit the wiki yourself. Doing so doesn't require extra permissions or anything.

**Comment by orbea** [comment]
Good point, I have never edited a github wiki before. I'll give it a try later. :)
